library/core/src/intrinsics/simd.rs

```diff
@@ -315,6 +333,18 @@ extern "platform-intrinsic" {
     /// Starting with the value `y`, add the elements of `x` and accumulate.
     pub fn simd_reduce_add_ordered<T, U>(x: T, y: U) -> U;
 
+    /// Add elements within a vector in arbitrary order, and without regard
+    /// for signed zeros.
+    ///
+    /// `T` must be a vector of integer or floating-point primitive types.
+    ///
+    /// `U` must be the element type of `T`.
+    ///
+    /// # Safety
+    ///
+    /// All input elements must be finite (i.e., not NAN and not +/- INF).
```

Comment:
Is this comment accurate? I assume there's a std::arch reason but I wouldn't have picked this intrinsic name

Reply:
This definitely shouldn't be using UB-causing fast-math flags without clearly having `_fast` in the name. Can we fix this intrinsic to only use the `reassoc` flag instead of the `fast` flag? This will eliminate the potential for UB.

Reply:
This will also fix https://github.com/rust-lang/stdarch/issues/1533.

Reply:
The comment is accurate wrt the current implementation in the LLVM codegen backend.

Even `reassoc` seems sketchy. That will allow LLVM to use re-associating optimization when multiple of these intrinsics are used in a row. I don't think this is a reasonable way to model the hardware operation.

What even is the Intel intrinsic doing here, why does it need a fast-math flag to begin with?